community-tutorials/mail-on-ssh-login/01-en.md

```diff
@@ -0,0 +1,165 @@
+
+- Username for the new created email address
+- Password for the new created email address
+- SMTP server for the new created email address
+- SMTP Port, should be 25.
+
+Create configuration file for SSMTP
+
+```
+sudo nano  /etc/ssmtp/ssmtp.conf
+```
+
+add the following lines but update the login information to work with your newly created mail account.
+
+```
+#/etc/ssmtp/ssmtp.conf
+root=server@YourDomainFromNetcup.com
+mailhub= mx2f11.netcup.net:25
+rewriteDomain=YourDomainFromNetcup.com
+AuthUser=YourDomainFromNetcup.com 
+AuthPass=YourPasswordForTheMailAccount
+FromLineOverride=YES
+```
+
+This will enbale SSMTP to send mails from your email account. It is recommended to create a mail account for this purpose only! Don’t use your daily account!
+
+# Step 4 – create a mail template
+The server is now able to write mails. To do that on a login, we must create a mail template.
+We could just send a simple alert, but it is nice to get more information to identify a possible thread.
+Therefore we will send a mail that contains the username of the user, the IP, date and time of the login. 
+
+Create a file for that purpose.
+`sudo nano /srv/mailatlogin.sh `
+
+Copy and paste this content to your file
+```
+#!/bin/sh
+date=$(date '+%d.%m.%Y')
+host=$(hostname)
+ip=$(hostname -I)
+{
+echo To: Reciever@yourDomain.com
+echo From: SERVER YourDomainFromNetcup.com
+echo Subject: "SERVER LOGIN! from $PAM_USER on $host IP: $PAM_RHOST $date"
+echo
+echo "Login detected on the server $HOSTNAME from 
+IP: $ip 
+User: $PAM_USER
+If this is not you, act immediately!"
+} | ssmtp Reciever@yourDomain.com
+```
+
+Save and exit the editor.
+The first lines reads the information from your system and stores them into variables for later use. We get the date, hostname and ip address. Then we will add this information to your mail.  
+All content between the curly braces is the mail.  
+Echo To  and Echo From must be supplemented with your data  
+Echo Subject: “ is the Subject of your mail. It can be edited if needed. All values starting with $ are variables. There will automatically be replaced with the corresponding value.  
+The next echo is the content of the mail and ca be edited or replaced.   
+Please be aware of the last line! Here you need to replace Reciever@yourDomain.com with your mailadress!
+# Step 5. Testing
+Test the script with the command
+```
+sudo /srv/mailatlogin.sh
+```
+And check your mail address. You should have gotten a new mail.
+# Step 6. Add the script to your sshd configuration
+Open the sshd.config with 
+```
+nano /etc/pam.d/sshd
+```
+and add the following line on the end
+```
+session optional pam_exec.so /srv/mailatlogin.sh
+```
+Save and close the file.
+Restart the ssh deamdon with
+```
+systemctl restart ssh
+```
+
+# Step 7. Relog
+Log our from your system with
+`exit`
+and login again.
+Check your mails! You should have gotten a new mail.
```

Comment:
After logging in, I get this error in the welcome message.
```
/srv/mailatlogin.sh failed: exit code 13
```

Reply:
Same here, should be because the script was not executeable.
Thanks for the feedback!